### PR TITLE
drivers: counter: nrf_rtc: Fix top value handling

### PR DESCRIPTION
**Pull request by nordic-krch**
Top value interrupt was not enabled because channel index was used instead of mask. Additionally, interrupt was enabled only when user callback was provided and not in case there was custom top value and no top callback.

### CODE REVIEW

File: drivers/counter/counter_nrfx_rtc.c [hunk 1]
```diff
@@ -141,6 +142,18 @@ static int counter_nrfx_cancel_alarm(struct device *dev, u8_t chan_id)
 	return 0;
 }
 
+/* Return true if counter must be cleared by the CPU. It is cleaned
+ * autonomously in case of max top value or PPI usage.
```
Comment (pabigot) [inline drivers/counter/counter_nrfx_rtc.c]: s/cleaned autonomously/cleared automatically/

note both words change.

**Review by pabigot — APPROVED** [review]
Minor wordsmithing in comment, otherwise fine.

**Comment by ioannisg** [comment]
Is this a bug @nordic-krch ?

**Comment by nordic-krch** [comment]
@ioannisg yes it is. Are you asking in the context of backporting to 2.0?

**Comment by ioannisg** [comment]
> 
> 
> @ioannisg yes it is. Are you asking in the context of backporting to 2.0?

Nope, just asking. It is nice if PR authors add bug labels, to indicate that this is a bug-fix attempt :)

**Comment by nordic-krch** [comment]
I thought that `bug` label is for issues with bug reports and PR that introduces bugs 😄

**Review by anangl — APPROVED** [review]

**Review by pizi-nordic — APPROVED** [review]

File: drivers/counter/counter_nrfx_rtc.c (hunk 1 above)
Comment (nordic-krch) [reply drivers/counter/counter_nrfx_rtc.c]: fixed

**Comment by nordic-krch** [comment]
@carlescufi @nashif someone can merge?

**Pull request merged**